src/mongo/registerMongoCommands.ts

```diff
@@ -44,23 +47,20 @@ export function registerMongoCommands(context: vscode.ExtensionContext, actionHa
         if (!node) {
             node = <IAzureParentNode<MongoDatabaseTreeItem>>await tree.showNodePicker(MongoDatabaseTreeItem.contextValue);
         }
-        if (connectedDb) {
-            connectedDb.treeItem.isConnected = false;
-            await connectedDb.refresh();
-        }
-        connectedDb = node;
-        await languageClient.connect(connectedDb.treeItem.connectionString, connectedDb.treeItem.databaseName);
-        connectedDb.treeItem.isConnected = true;
-        await node.refresh();
+
+        await languageClient.connect(node.treeItem.connectionString, node.treeItem.databaseName);
         context.globalState.update(connectedDBKey, node.id);
+        ext.connectedMongoDB = node;
+
+        await tree.refresh(node.parent);
```

Comment:
Don't think this will handle the case where you're switching the connection from an Azure node to an attached node?

Reply:
Good point